appjs/views/edit_project.js

```diff
@@ -170,85 +210,113 @@ var EditProject = BaseView.extend(require('./mixins/actions'), require('./mixins
+      this.previousData = data;
 
-    // stash data so we can see if it changed
-    this.previousData = data;
+      // Now that data is connected and valid, show some sort of loading indicator:
+      if($('#embed-preview.validation-error')){
+        $('#embed-preview').removeClass('validation-error').addClass('loading');
+      }
 
-    // Show some sort of loading indicator:
-    $('#embed-preview').addClass('loading');
+      return $.ajax({
+        type: "POST",
+        url: this.model.url() + "/preview_build_data" + query,
+        data: JSON.stringify(data),
+        contentType: 'application/json',
+        dataType: 'json'
+      }).then(function( data ) {
+        logger.debug('Updating live preview...');
+        var iframeLoaded = _.once(function() {
+          view.pym.sendMessage('updateData', JSON.stringify(data));
+          $('#embed-preview').removeClass('loading');
+        });
 
-    return $.ajax({
-      type: "POST",
-      url: this.model.url() + "/preview_build_data" + query,
-      data: JSON.stringify(data),
-      contentType: 'application/json',
-      dataType: 'json'
-    }).then(function( data ) {
-      logger.debug('Updating live preview...');
-      var iframeLoaded = _.once(function() {
-        view.pym.sendMessage('updateData', JSON.stringify(data));
-        $('#embed-preview').removeClass('loading');
-      });
+        if(data.theme !== view.theme || !view.pym){
+          if(typeof data.theme !== 'undefined'){
+            view.theme = data.theme;
+            view.getTwitterCount();
+          }
+          var version = view.model.getVersion(),
+            previewSlug = view.model.isThemeable() ?
+                version :[version, view.theme].join('-'),
+            previewUrl = view.model.blueprint.getMediaUrl( previewSlug + '/preview');
 
-      if(data.theme !== view.theme){
-        if(typeof data.theme !== 'undefined'){
-          view.theme = data.theme;
+          if ( view.pym ) { view.pym.remove(); }
+          view.pym = new pym.Parent('embed-preview', previewUrl);
+          view.pym.iframe.onload = iframeLoaded;
+
+          // In case some dumb script hangs the loading process
+          setTimeout(iframeLoaded, 20000);
+        } else {
+          iframeLoaded();
         }
-        var version = view.model.getVersion(),
-          previewSlug = view.model.isThemeable() ?
-              version :[version, view.theme].join('-'),
-          previewUrl = view.model.blueprint.getMediaUrl( previewSlug + '/preview');
+      }, function(err) {
+        if ( err.status < 500 ) {
+          view.app.view.error(
+            "Can't update live preview (" +err.responseJSON.error+").", 'permanent');
+        } else {
+          view.app.view.error(
+            "Could not update live preview, please contact support.", 'permanent' );
+          logger.error(err);
+        }
+      });
+    }
+  }, 500),
 
-        if ( view.pym ) { view.pym.remove(); }
+  getTwitterCount: function(){
```

Comment:
Sets the available character count for share text.
![screen shot 2016-07-05 at 4 43 04 pm](https://cloud.githubusercontent.com/assets/3526984/16599277/98822548-42cf-11e6-9ad2-dfd5ff557733.png)